mpi/distribute_flux_sparsity_pattern.mpirun=2.release fails

looks serious:
```
$ ctest -V -R "mpi/distribute_flux_sparsity_pattern.mpirun=2.release"
UpdateCTestConfiguration  from :/home/davydden/libs/dealii-tests/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/davydden/libs/dealii-tests/DartConfiguration.tcl
Test project /home/davydden/libs/dealii-tests
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 6024
    Start 6024: mpi/distribute_flux_sparsity_pattern.mpirun=2.release

6024: Test command: /home/davydden/spack/opt/spack/linux-ubuntu16-x86_64/gcc-6.3.0/cmake-3.7.2-qfd24w4jz4yvsvcy5foddwgiwyykqkuq/bin/cmake "-DTRGT=distribute_flux_sparsity_pattern.mpirun2.release.diff" "-DTEST=mpi/distribute_flux_sparsity_pattern.mpirun=2.release" "-DEXPECT=PASSED" "-DBINARY_DIR=/home/davydden/libs/dealii-tests/mpi" "-DGUARD_FILE=/home/davydden/libs/dealii-tests/mpi/distribute_flux_sparsity_pattern.release/interrupt_guard.cc" "-P" "/home/davydden/spack/opt/spack/linux-ubuntu16-x86_64/gcc-6.3.0/dealii-develop-wrha3kpxcvgg2wd54rjm5dxl6ksdihf3/share/deal.II/scripts/run_test.cmake"
6024: Test timeout computed to be: 600
6024: make[3]: *** [distribute_flux_sparsity_pattern.release/mpirun=2/output] Error 1
6024: make[2]: *** [CMakeFiles/distribute_flux_sparsity_pattern.mpirun2.release.diff.dir/all] Error 2
6024: make[1]: *** [CMakeFiles/distribute_flux_sparsity_pattern.mpirun2.release.diff.dir/rule] Error 2
6024: make: *** [distribute_flux_sparsity_pattern.mpirun2.release.diff] Error 2
6024: Test mpi/distribute_flux_sparsity_pattern.mpirun=2.release: RUN
6024: ===============================   OUTPUT BEGIN  ===============================
6024: Built target distribute_flux_sparsity_pattern.release
6024: Generating distribute_flux_sparsity_pattern.release/mpirun=2/output
6024: mpi/distribute_flux_sparsity_pattern.mpirun=2.release: BUILD successful.
6024: mpi/distribute_flux_sparsity_pattern.mpirun=2.release: RUN failed. ------ Return code 59
6024: mpi/distribute_flux_sparsity_pattern.mpirun=2.release: RUN failed. ------ Result: /home/davydden/libs/dealii-tests/mpi/distribute_flux_sparsity_pattern.release/mpirun=2/failing_output
6024: mpi/distribute_flux_sparsity_pattern.mpirun=2.release: RUN failed. ------ Partial output:
6024: 
6024: mpi/distribute_flux_sparsity_pattern.mpirun=2.release: RUN failed. ------ Additional output on stdout/stderr:
6024: 
6024: [0]PETSC ERROR: [1]PETSC ERROR: ------------------------------------------------------------------------
6024: [0]PETSC ERROR: Caught signal number 11 SEGV: Segmentation Violation, probably memory access out of range
```

debug versions passes. 

This happens with `gcc@6.3.0` and `openmpi@2.0.2`. For the record, **all other 8678 tests but two opencascade related https://github.com/dealii/dealii/issues/3915 pass!**

--- a/tests/mpi/distribute_flux_sparsity_pattern.cc
+++ b/tests/mpi/distribute_flux_sparsity_pattern.cc
@@ -231,7 +231,7 @@ namespace LinearAdvectionTest
                 if (neighbor_index != -1) // interior face
                   {
                     typename DoFHandler<dim>::active_cell_iterator neighbor_cell =
-                      neighbor_cell = current_cell->neighbor(face_n);
+                      current_cell->neighbor(face_n);
 
                     bool do_face_integration = false;
                     bool neighbor_is_level_lower = false;

